x/slashing/handler_sidechain.go

```diff
@@ -31,23 +28,16 @@ func handleMsgBscSubmitEvidence(ctx sdk.Context, msg MsgBscSubmitEvidence, k Kee
 
 	if sdk.IsUpgrade(sdk.BEP174) {
 		sideConsAddr, err = msg.Headers[0].ExtractSignerFromHeader(chainID)
-	} else {
-		sideConsAddr, err = msg.Headers[0].ExtractSignerFromHeader(nil)
-	}
-	if err != nil {
-		return ErrInvalidEvidence(DefaultCodespace, fmt.Sprintf("Failed to extract signer from block header, %s", err.Error())).Result()
-	}
-	if sdk.IsUpgrade(sdk.BEP174) {
 		sideConsAddr2, err = msg.Headers[1].ExtractSignerFromHeader(chainID)
 	} else {
 		sideConsAddr, err = msg.Headers[0].ExtractSignerFromHeader(nil)
+		sideConsAddr2, err = msg.Headers[1].ExtractSignerFromHeader(nil)
```

Comment:
The `err` will be overwritten here.  We should have `err` and `err2` here.

Reply:
ok